clippy_lints/src/methods/collapsible_str_replace.rs

```diff
@@ -0,0 +1,160 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::get_parent_expr;
+use clippy_utils::visitors::for_each_expr;
+use clippy_utils::SpanlessEq;
+use core::ops::ControlFlow;
+use if_chain::if_chain;
+use rustc_ast::ast::LitKind;
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_hir::{ExprKind, Path, QPath};
+use rustc_lint::LateContext;
+use rustc_span::source_map::Spanned;
+use rustc_span::Span;
+
+use super::method_call;
+use super::COLLAPSIBLE_STR_REPLACE;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'tcx>, to: &'tcx hir::Expr<'tcx>) {
+    if_chain! {
+        if let Some(replace_methods) = collect_replace_calls(cx, expr, to);
+        if let Some(earliest_replace_call) = replace_methods.methods.last();
```

Comment:
You should use a `Deque` instead of a `Vec` and use `push_front`, so that the logic is a bit cleaner.